modules/product/src/main/java/com/opengamma/strata/product/swap/IborRateCalculation.java

```diff
@@ -319,7 +322,9 @@ public void collectIndices(ImmutableSet.Builder<Index> builder) {
     // resolve against reference data once
     DateAdjuster fixingDateAdjuster = fixingDateOffset.resolve(refData);
     Function<SchedulePeriod, Schedule> resetScheduleFn =
-        getResetPeriods().map(rp -> rp.createSchedule(accrualSchedule.getRollConvention(), refData)).orElse(null);
+        getResetPeriods().map(rp -> getIndex().getCurrency().equals(CNY) && rp.getResetFrequency().equals(P1W) ?
```

Comment:
I think we can be more general here. If `accrualSchedule.getFrequency().isMonthBased()` and `rp.getResetFrequency().isWeekBased()` then...